core/src/main/java/org/mskcc/cbio/oncokb/util/EvidenceUtils.java

```diff
@@ -613,24 +613,57 @@ public static Oncogenicity getOncogenicityFromEvidence(Set<Evidence> evidences)
         return levels;
     }
 
+    private static Map<String, Object> getBiggerItem(List<String> query, Set<List<String>> keys) {
+        Map<String, Object> map = new HashMap<>();
+        map.put("exist", false);
+        map.put("origin", false);
+        map.put("result", query);
+        for(List<String> key : keys) {
+            // Check whether key has all elements in query;
+            if (key.containsAll(query)) {
+                map.put("exist", true);
+                map.put("result", key);
+                return map;
+            }
+
+            // Check whether query has all elements in key;
+            if (query.containsAll(key)) {
+                map.put("exist", true);
+                map.put("origin", true);
+                map.put("result", key);
+                return map;
+            }
+        }
+        return map;
+    }
+
     public static Set<Evidence> keepHighestLevelForSameTreatments(Set<Evidence> evidences) {
-        Map<String, Set<Evidence>> maps = new HashedMap();
+        Map<List<String>, Set<Evidence>> maps = new HashedMap();
         Set<Evidence> filtered = new HashSet<>();
 
         for (Evidence evidence : evidences) {
             if (evidence.getTreatments() != null && evidence.getTreatments().size() > 0) {
-                String treatmentsName = TreatmentUtils.getTreatmentName(new HashSet<>(evidence.getTreatments()));
-                if (!maps.containsKey(treatmentsName)) {
-                    maps.put(treatmentsName, new HashSet<Evidence>());
+                List<String> treatments = TreatmentUtils.getTreatments(new HashSet<>(evidence.getTreatments()));
+
+                Map<String, Object> map = getBiggerItem(treatments, maps.keySet());
+                if ((boolean) map.get("exist")) {
```

Comment:
Just curious. Why do you convert it forcefully to boolean in here? I thought its value would only be True or False.

Reply:
The map has Object content, you need to force converting Object to boolean in order to pass the validation process.